cwltool/command_line_tool.py

```diff
@@ -1495,8 +1495,21 @@ def collect_output(
                             )
                         primary["format"] = format_eval
                 else:
-                    for primary in aslist(result):
-                        primary["format"] = format_field
+
+                    def recursively_insert(j_dict: Any, key: Any, val: Any) -> Any:
+                        """Recursively insert a value into any dictionary."""
+                        if isinstance(j_dict, List):
+                            return [recursively_insert(x, key, val) for x in j_dict]
+                        if isinstance(j_dict, Dict):
```

Comment:
```suggestion
                        if isinstance(j_dict, MutableSequence):
                            return [recursively_insert(x, key, val) for x in j_dict]
                        if isinstance(j_dict, MutableMapping):
```

https://docs.python.org/3.8/library/typing.html#typing.Dict

> To annotate arguments it is preferred to use an abstract collection type such as [`Mapping`](https://docs.python.org/3.8/library/typing.html#typing.Mapping).

https://docs.python.org/3.8/library/typing.html#typing.List

> To annotate arguments it is preferred to use an abstract collection type such as [`Sequence`](https://docs.python.org/3.8/library/typing.html#typing.Sequence) or [`Iterable`](https://docs.python.org/3.8/library/typing.html#typing.Iterable).

Reply:
Thanks, I will keep that in mind in future. Fixed.